### PR TITLE
PWN-7451 - Various Swap fixes

### PR DESCRIPTION
**Pull request by 1sakov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-7451
https://p2pvalidator.atlassian.net/browse/PWN-7456
https://p2pvalidator.atlassian.net/browse/PWN-7458
https://p2pvalidator.atlassian.net/browse/PWN-7481
https://p2pvalidator.atlassian.net/browse/PWN-7484

## Description of Work

[ what work was done in this ticket ]

## Screenshots (Optional)

| Screenshot #1 | Screenshot #2 | Screenshot #3 |
|:-------------:|:-------------:|:-------------:|
| [Paste Pic 1] | [Paste Pic 2] | [Paste Pic 3] |

### CODE REVIEW

**Review by NikolaiEmelianov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/token_selector/CommonSwapTokenSelector.kt [hunk 1]
```diff
@@ -55,10 +56,10 @@ class CommonSwapTokenSelector(
                     findSolOrUsdc = false
                 )
             }
-            // not have any tokens
+            // no any tokens
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/jupiter/statemanager/token_selector/CommonSwapTokenSelector.kt:59]: // user has no active tokens

maybe better?

File: app/src/main/res/layout/fragment_jupiter_swap.xml [hunk 2]
```diff
@@ -181,5 +181,51 @@
...
+        android:gravity="center"
+        android:orientation="vertical"
+        android:visibility="gone">
+
+        <Space
+            android:layout_width="wrap_content"
+            android:layout_height="0dp"
+            android:layout_weight="1" />
+
+        <TextView
```
Comment (gslevinkov) [inline app/src/main/res/layout/fragment_jupiter_swap.xml:197]: auto QA asked us to put IDs on views, maybe add them here too?

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt [hunk 3]
```diff
@@ -74,6 +77,9 @@ class SwapTokensPresenter(
                 val searchResult = interactor.searchToken(tokenToChange, newQuery)
                 val cellItems = searchResultMapper.toCellItems(searchResult)
                 view?.setTokenItems(cellItems)
+
+                val isEmpty = cellItems.none { it is FinanceBlockCellModel }
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt:81]: isEmpty - isSearchResultEmpty?

File: app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt [hunk 4]
```diff
@@ -57,7 +58,9 @@ class SwapTokensPresenter(
                 )
             }
         }
+        // todo: optimize. There are about 800~ items being set at once, we need to set list partially
```
Comment (gslevinkov) [inline app/src/main/java/org/p2p/wallet/swap/ui/jupiter/tokens/presenter/SwapTokensPresenter.kt:61]: is this a perfomance problem? i thought recyclerview handles this via viewholders

File: app/src/main/res/values/swap_strings.xml [hunk 5]
```diff
@@ -27,6 +24,7 @@
     <string name="swap_tokens_section_chosen_token">CHOSEN TOKEN</string>
     <string name="swap_tokens_section_all_tokens">ALL TOKENS</string>
     <string name="swap_tokens_popular_label">popular</string>
+    <string name="swap_tokens_you_pay_only">🤖 You can enter "You pay" field only</string>
```
Comment (gslevinkov) [inline app/src/main/res/values/swap_strings.xml:27]: \\"You pay\\"

**Review by gslevinkov — APPROVED** [review]

**Pull request merged**